segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ZKSegmentContainerMonitor.java

```diff
@@ -106,9 +107,16 @@ public void initialize() {
     public void initialize(Duration monitorInterval) {
         Exceptions.checkNotClosed(closed.get(), this);
 
+        // Start loading the segment container to node assigment map from zookeeper.
         this.hostContainerMapNode.start();
+
+        // There are two triggers for the segment container monitor.
+        // 1. On any segment container ownership changes notified via zookeeper. We will ensure the local containers
+        //      are stopped/started according to the new ownership mapping.
+        // 2. At scheduled intervals to perform retries on local segment container start failures.
         this.assigmentTask.set(this.executor.scheduleWithFixedDelay(
                 this::checkAssignment, 0L, monitorInterval.getSeconds(), TimeUnit.SECONDS));
+        this.hostContainerMapNode.getListenable().addListener(this::checkAssignment, this.executor);
```

Comment:
I suppose that this is going to have a thread executing `checkAssignment` every time there is a change in zk? is it a better approach to queue the changes and execute them in order by a single thread rather than having the threads competing for the `checkAssignment` lock?